exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/OperatorWrapper.java

```diff
@@ -130,4 +136,32 @@ public void addSummary(TableBuilder tb) {
     tb.appendBytes(Math.round(memSum / size), null);
     tb.appendBytes(peakMem.getLeft().getPeakLocalMemoryAllocated(), null);
   }
+
+  public String getMetricsTable() {
+    if (!OperatorMetricRegistry.contains(operatorType.getNumber())) {
+      return "";
+    }
+    final ArrayList<String> metricNames = Lists.newArrayList("Minor Fragment");
+    for (final MetricValue metric : firstProfile.getMetricList()) {
+      metricNames.add(OperatorMetricRegistry.getMetricName(operatorType.getNumber(), metric.getMetricId()));
+    }
+
+    final String[] metricsTableColumnNames = new String[metricNames.size()];
+    final TableBuilder builder = new TableBuilder(metricNames.toArray(metricsTableColumnNames));
+    for (final ImmutablePair<OperatorProfile, Integer> ip : ops) {
+      final OperatorProfile op = ip.getLeft();
+
+      builder.appendCell(
+          new OperatorPathBuilder()
+              .setMajor(major)
+              .setMinor(ip.getRight())
+              .setOperator(op)
+              .build(),
+          null);
+      for (final MetricValue metric : op.getMetricList()) {
+        builder.appendInteger(metric.getLongValue(), null);
```

Comment:
the code builds the table header from the first profile's metric list, then assumes all minor fragments will have the metrics in the same order of the first profile.

Can you confirm that this assumption is always true ?